### DISCUSSION TITLE
can‘t load twin.macro in MDX files

### DISCUSSION THREAD

**Discussion by ajmalafif** [ben-rogerson/twin.macro#385] [Q&A]
Hi,

I can‘t seem to figure out why `import 'twin.macro'` isn‘t working when importing: https://raw.githubusercontent.com/ajmalafif/afif.dev/twinmacro-as-shortcodes-for-mdx-afif-521/web/src/content/projects/fare-alert.mdx

 I‘ve tried few other ways like importing as `shortcodes` like via `Layout` or `GlobalStyles`:  https://github.com/ajmalafif/afif.dev/blob/9a1ac57b6cd461fe1e004f063c849d5ca6487026/web/src/templates/project-post.js#L19
 
 ```
 const shortcodes = { Link, Hidden, Preview, Layout, pre: CodeBlock } 
 ```
 
 Nothing seems to work. Any pointer or advice?
 
 edit: reckon it has something to do with this? https://www.gatsbyjs.com/docs/using-client-side-only-packages/

**Comment by amantiwari1** [comment]
try it 
```js
/** @jsxImportSource @emotion/react */
// eslint-disable-next-line
import tw from 'twin.macro'
```

**Reply by ajmalafif** [reply]
> ```js
> /** @jsxImportSource @emotion/react */
> // eslint-disable-next-line
> import tw from 'twin.macro'
> ```

hey @amantiwari1 appreciate the help, I am getting this error instead:
```
./src/templates/project-post.js
Syntax error: /web/src/templates/project-post.js: importSource cannot be set when runtime is classic.
> 1 | /** @jsxImportSource @emotion/react */
    | ^
  2 | // eslint-disable-next-line
  3 | import React from 'react'
  4 | import Helmet from 'react-helmet'
  ```
  
<img width="1259" alt="Screenshot 2021-04-13 at 12 38 58 AM" src="https://user-images.githubusercontent.com/827167/114430226-cd139100-9bf0-11eb-8ea4-b376576d03c4.png">

**Reply by amantiwari1** [reply]
sorry i thought you used emotion library, any way I don't know

**Comment by bumi001** [comment]
Not sure why you would want to import tw in a mdx file. However, I am able to import a jsx component in a mdx file where the jsx component uses tw. I am using the following in my webpack.config.js:
      {
        test: /\.mdx?$/i,
        use: [{
          loader: '@mdx-js/loader',
          options: {
            jsxImportSource: '@emotion/react'
          }
        }],
      },


See also: https://giters.com/ben-rogerson/twin.macro/issues/589
